GitCommands/Git/GitModule.cs

```diff
@@ -2125,15 +2125,10 @@ public IEnumerable<string> GetSettings(string setting)
             return LocalConfigFile.GetValues(setting);
         }
 
-        public string GetSetting(string setting)
-        {
-            return LocalConfigFile.GetValue(setting);
-        }
+        public string GetSetting(string setting) => LocalConfigFile.GetValue(setting);
 
-        public string GetEffectiveSetting(string setting)
-        {
-            return EffectiveConfigFile.GetValue(setting);
-        }
+        public string GetEffectiveSetting(string setting) => EffectiveConfigFile.GetValue(setting);
+        public bool IsEffectiveSettingEnabled(string setting) => EffectiveConfigFile.IsEnabled(setting);
```

Comment:
```suggestion
        public bool? GetEffectiveBoolSetting(string setting) => EffectiveConfigFile.GetBool(setting);
```

Reply:
I'd like us to consider an opportunity to generalise and have something like:
```cs
public T GetEffectiveSetting<T>(string settingName) 
{
   ...
}

public T GetSetting<T>(string settingName) 
{
   ...
}
```